Replaced react-datepicker with native HTML datepicker

glower opened the pull request:
- Replaced react-datepicker with native HTML datepicker on Create System-User page
- Replaced moment.js with native js date object
- Added missing tests for "Create System-User" page
- Added golang test for the date-time format

glower commented:
After bumping the version of react and react-datepicker we broke /systemuser page. This happened because react-datepicker was not using moment.js anymore (https://www.npmjs.com/package/react-datepicker#momentjs)

With this change we will use native js Date object to fix the broken datepicker.
The change will also add missing test for this page

anthonydillon commented on webapp-admin/src/components/SystemUserForm.js:
```diff
@@ -21,19 +21,28 @@ import Assertion from '../models/assertions'
         super(props);
 
+    	// same as moment.utc()
+    	let startDate = new Date(new Date().toUTCString());
```
These can be `const` as they are not reassigned.

anthonydillon commented on webapp-admin/src/components/SystemUserForm.js:
```diff
@@ -21,19 +21,28 @@ import Assertion from '../models/assertions'
+    	// same as moment.utc()
+    	let startDate = new Date(new Date().toUTCString());
+    	// same as moment.utc().add(1, "year")
+    	let endDate = new Date(new Date().toUTCString()).setFullYear(
```
Same here

glower replied:
fixed

anthonydillon commented on webapp-admin/src/components/SystemUserForm.js:
```diff
@@ -215,7 +224,7 @@ class SystemUserForm extends Component {
                             <div className="row">
                                 <div className="col-3">
-                                    <DatePicker selected={this.state.since} onChange={this.handleChangeSinceDate} />
+                                    <DatePicker selected={this.state.startDate} onChange={this.handleChangeSinceDate} />
```
Since handleChangeSinceDate function is updating state.since the selected date will not be updated unless this works by being the last thing the user selected.

glower commented:
* <DatePicker> + <input type="time">  is replaced with <input type="datetime-local"> element where you can set both: date and time with one element.
* onChange methods are now more simple

glower commented:
> Sorry I'm not sure what is the acturally problem here but was asked if someone could take a look. Would it be easier to use the native HTML datepicker if your not using moment:
> 
> ```
> <input type="date">
> ```

Thanks for the hint, I just replaced it with the native datepicker, it make more sense now and the code is simple now. 
I also added a backend test to test that the new generated format works there (in the end it's the same date-time string)

anthonydillon commented on webapp-admin/package.json:
```diff
@@ -8,7 +8,6 @@
     "moment": "^2.24.0",
     "npm": "^6.13.6",
     "react": "^16.12.0",
-    "react-datepicker": "^2.11.0",
```
:+1:

anthonydillon replied:
Always nice to remove deps
